core-bundle/contao/modules/ModulePersonalData.php

```diff
@@ -64,7 +64,7 @@ public function generate()
 			return '';
 		}
 
-		if (!$security->isGranted('IS_AUTHENTICATED_FULLY'))
+		if ($this->reqFullAuth && !$security->isGranted('IS_AUTHENTICATED_FULLY'))
```

Comment:
```suggestion
		if (($this->reqFullAuth || \in_array('password', $this->editable, true)) && !$security->isGranted('IS_AUTHENTICATED_FULLY'))
```
Should we force the requirement for authenticated_fully if the password field is present?

Reply:
I would not do this automatically, as it could confuse users if the checkbox is switched off and the function is still activated.